typegraph/core/src/utils/mod.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use std::collections::HashMap;
+
+use crate::{
+    errors::Result,
+    global_store::with_store,
+    wit::core::{Core, TypeBase, TypeId, TypeStruct, TypeWithInjection},
+    Lib,
+};
+
+mod apply;
+
+fn find_missing_props(
+    old_props: &Vec<(String, u32)>,
+    new_props: &Vec<(String, u32)>,
+) -> Vec<(String, u32)> {
+    let mut missing_props = vec![];
+    for (k_old, v_old) in old_props {
```

Comment:
use hashset from keys and set operation?

Reply:
We only have to test against the names. That could work if the pairs (name, id) in the two vecs (old and new) always overlap. In this case, ids can differ even if the names are the same.